osidb/helpers.py

```diff
@@ -238,3 +240,98 @@ def safe_get_response_content(response: Response):
+        return encoded_json
+
+    def makeSocket(self, *args, **kwargs):
+        """
+        A factory method which allows subclasses to define the precise
+        type of socket they want with SSL support.
+        """
+
+        result = super().makeSocket(*args, *kwargs)
+
+        context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+        context.load_verify_locations(
+            cafile="/opt/app-root/etc/logstash/certs/osidb-logstash-ca.crt"
+        )
+        context.load_cert_chain(
+            certfile="/opt/app-root/etc/logstash/certs/osidb-logstash.crt",
+            keyfile="/opt/app-root/etc/logstash/certs/osidb-logstash.key",
+        )
+
+        result = context.wrap_socket(result, server_hostname=self.address[0])
+        return result
+
+    # def send(self, s):
+    #     print(self.sock)
+    #     if self.sock is None:
+    #         self.createSocket()
+    #     # self.sock can be None either because we haven't reached the retry
+    #     # time yet, or because we have reached the retry time and retried,
+    #     # but are still unable to connect.
+    #     # print(self.sock._cl)
+
+    #     if self.sock:
+    #         print("really really sending")
+    #         self.sock.sendall(s)
+    #         # response = self.sock.recv(1024)
+    #         # print(response)
+
+    #         # try:
+    #         #     self.sock.sendall(s)
+    #         # except OSError:  # pragma: no cover
+    #         #     logging.error("error error")
+    #         #     self.sock.close()
+    #         #     self.sock = None  # so we can call createSocket next time
+    #     else:
+    #         raise socket.error
+
+    # def emit(self, record):
+    #     try:
+    #         s = self.makePickle(record)  # Serialize the log record
+    #         for attempt in range(self.retries):
+    #             try:
+    #                 print("sending")
+
+    #                 self.send(s)  # Attempt to send the serialized log message
+    #                 return  # Exit if send is successful
+    #             except (socket.error, ssl.SSLError, BrokenPipeError):
+    #                 print("retrying")
+    #                 if attempt < self.retries - 1:
+    #                     time.sleep(self.delay)  # Wait before retrying
+    #                 else:
+    #                     # Log to the fallback file after all retries fail
+    #                     if self.fallback_handler is not None:
+    #                         print("fallback")
+    #                         self.fallback_handler.emit(record)  # Fallback to file
+    #                     # self.handleError(record)  # Handle error logging
+    #     except Exception as e:
+    #         # Handle unexpected exceptions and log to fallback file
+    #         if self.fallback_handler is not None:
+    #             print("fallback")
+    #             self.fallback_handler.emit(record)
```

Comment:
Is there any purpose of leaving these commented lines here?

Reply:
I see they are being removed in the following commit. Then it makes even less sense to me to introduce them here but at least they are not part of the final result. I have no strong preference then.

Reply:
Seems like I messed up rebasing :smile:  I will fix that, thanks!